core/spec/models/spree/concerns/user_address_book_spec.rb

```diff
@@ -242,6 +242,35 @@ module Spree
       it "returns false if the addresses is not there" do
         expect(user.remove_from_address_book(0)).to be false
       end
+
+      context 'when user has previous order addresses' do
```

Comment:
I think that besides all these cases when the attributes match and so they are then changed to `nil`, it would be nice to have at least one test for the other scenarios when they don't match and they are not changed... what do you think?

Reply:
Hey @spaghetticode, thank you for the comment! 
Added example when the address does not match any user addresses: https://github.com/solidusio/solidus/pull/3482/files#diff-e930e6c49d94b00d1756dfaf242e4d68R252-R264